src/utils/ContractUtils.ts

```diff
@@ -146,8 +146,8 @@ export async function amountToUnits(
   amount: string
 ): Promise<string> {
   try {
-    const tokenContract = new web3.eth.Contract(minAbi, token)
-    let decimals = await tokenContract.methods.decimals().call()
+    const tokenContract = token && new web3.eth.Contract(minAbi, token)
+    let decimals = token ? await tokenContract.methods.decimals().call() : '0'
```

Comment:
what weird logic combined with next 3 lines which should be all removed. If `decimals` is set here to `'0'` as part of `else` then no need to check for it again in next lines to set it to something else again. So for `else`, `decimals` here should be just `18` and then no need for the next 3 lines. And then it's a `const` and no `let`